emcc.py

```diff
@@ -2031,7 +2031,10 @@ def check_memory_setting(setting):
     if options.shell_path == shared.path_from_root('src', 'shell.html'):
       options.shell_path = shared.path_from_root('src', 'shell_minimal_runtime.html')
 
-    if settings.ASSERTIONS and settings.MINIMAL_RUNTIME:
+    if settings.EXIT_RUNTIME:
+      settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE += ['proc_exit']
+
+    if settings.ASSERTIONS:
       # In ASSERTIONS-builds, functions UTF8ArrayToString() and stringToUTF8Array() (which are not JS library functions), both
       # use warnOnce(), which in MINIMAL_RUNTIME is a JS library function, so explicitly have to mark dependency to warnOnce()
```

Comment:
This PR changes the logic to remove MINIMAL_RUNTIME only check for adding `$warnOnce` (which I suppose is ok), but the comment still reads MINIMAL_RUNTIME specific. Although I suppose technically the comment is still right. Maybe it makes sense to adjust the comment to highlight the fact that `$warnOnce` is now added always for simplicity (I presume that is the rationale here?) even though needed just for MINIMAL_RUNTIME.

This change does have the effect of emitting `function warnOnce()` twice to non-MINIMAL_RUNTIME builds though, once via the preamble and a second time via the JS function. Not sure how well that will DCE via Closure, since it has problems optimizing away stuff from global JS scope.

Reply:
Sorry I should have mentioned that this entire block is already wrapped in `if MINIMAL_RUNTIME` (see line 2029) so this was a redundant check.   Its not obvious when reading the diff